src/demo-app/stepper/stepper-demo.html

```diff
@@ -1,3 +1,50 @@
+<h2>Linear Vertical Stepper Demo</h2>
+<form [formGroup]="formGroup">
+  <md-vertical-stepper formArrayName="formArray">
+    <md-step formGroupName="0">
+      <ng-template mdStepLabel>Fill out your name</ng-template>
+      <md-input-container>
+        <input mdInput placeholder="First Name" formControlName="firstNameFormCtrl" required>
+        <md-error>This field is required</md-error>
+      </md-input-container>
+
+      <md-input-container>
+        <input mdInput placeholder="Last Name" formControlName="lastNameFormCtrl" required>
+        <md-error>This field is required</md-error>
+      </md-input-container>
+      <div>
+        <button md-button mdStepperNext type="button">Next</button>
+      </div>
+    </md-step>
+
+    <md-step formGroupName="1"
+             [disabled]="formArray.hasError('invalid step') ?
+             formArray.getError('invalid step').index <= 1 : false">
+      <ng-template mdStepLabel>
+        <div>Fill out your phone number</div>
+      </ng-template>
+      <md-input-container>
+        <input mdInput placeholder="Phone number" formControlName="phoneFormCtrl">
+        <md-error>This field is required</md-error>
+      </md-input-container>
+      <div>
+        <button md-button mdStepperPrevious type="button">Back</button>
+        <button md-button mdStepperNext type="button">Next</button>
+      </div>
+    </md-step>
+
+    <md-step
+        [disabled]="formArray.hasError('invalid step') ?
```

Comment:
@g1shin I see that, but I think it's too complex for the end user to have to add that to each step. It's going to get confusing for people when you use that with `*ngFor` and people will make mistakes. Especially when it's going to have a lot more options added in future for `[editable]` and `[optionial]`. 

You should be using less than operator `<` instead of less than or equal operator `<=` for your disabled logic. The second step should be `[disabled]="formArray.hasError('invalid step') ?
             formArray.getError('invalid step').index < 1 : false"` you are also missing the required validation on your second step to be able to test this.

```
 phoneFormCtrl: ['', Validators.required],
```

Do you have the design document for the stepper? I don't see it on the [wiki](https://github.com/angular/material2/wiki/Design-doc-directory).